addons/account_edi_facturx/models/account_edi_format.py

```diff
@@ -133,11 +133,16 @@ def format_monetary(number, currency):
                 is_refund=line.move_id.move_type in ('in_refund', 'out_refund'),
             )
 
+            if line.discount == 100.0:
+                gross_price_subtotal = line.currency_id.round(line.price_unit * line.quantity)
+            else:
+                gross_price_subtotal = line.currency_id.round(line.price_subtotal / (1 - (line.discount / 100.0)))
```

Comment:
and with if discount is 100.0 or quantity 0.0 (to fix later)

Reply:
If discount is 100 and quantity is 0 (why would they do that) it would just be 0.
I'm not sure about what you mean

Reply:
if discount = 100%, gross_price_subtotal = 0.0 with quantity = 0.0 (ok)
else: price_subtotal = 0.0 since quantity = 0.0 so gross_price_subtotal = 0.0 (ok)
=> no issue with quantity = 0.0 then